ladon.go

```diff
@@ -102,21 +126,32 @@ func (l *Ladon) DoPoliciesAllow(r *Request, policies []Policy) (err error) {
 		}
 
 		// Is the policies effect deny? If yes, this overrides all allow policies -> access denied.
+		// Should this be not the first in the loop since it overrides all allow policies?
 		if !p.AllowAccess() {
 			deciders = append(deciders, p)
 			l.auditLogger().LogRejectedAccessRequest(r, policies, deciders)
+			go l.Metric.RequestDeniedBy(*r, p)
 			return errors.WithStack(ErrRequestForcefullyDenied)
 		}
 
 		allowed = true
+
 		deciders = append(deciders, p)
 	}
 
 	if !allowed {
+		// We return no match
+		go l.Metric.RequestNoMatch(*r)
+
 		l.auditLogger().LogRejectedAccessRequest(r, policies, deciders)
 		return errors.WithStack(ErrRequestDenied)
 	}
 
+	// We return first policy that matched
+	if len(deciders) > 0 {
+		l.Metric.RequestAllowedBy(*r, deciders[0])
```

Comment:
Wouldn't it make sense to include all the policies that caused this decision?

Reply:
This is quite interesting to discuss 😇 

So what I noticed is that whenever one policy does *deny* we just stop processing and return. Which is great - works like in AWS. 

Now for the allow policy I had the idea that I would just return the single one that decided. But I guess it makes sense to return all and decided client implementing it how to handle it right ?

Reply:
So what can happen is that we get 3 policies, two of them are allow, and one is deny. Deny overrules always! In that case, it would be interesting to know that actually some policies allowed, but that one denied it. Then it might be easier to debug, for example.

For allow, it's true - one is actually enough. But maybe we want to know simply all the policies that allow? That way we could, for example, remove all the policies from a subject that has allow.

Reply:
I have included all policies on `RequestAllowedBy` so this way we can know what was going on.